### ISSUE TITLE
Doesn't work without empty line breaks

### ISSUE THREAD

**Issue by Jinjiang** [koka831/remark-custom-container#210]
input:

```
normal body

::: info first

first container body

:::

rest element
```

output:

```
<p>normal body</p>
<div class="remark-container info"><div class="remark-container__title">first</div><p>first container body</p></div>
<p>rest element</p>
```

input2:

```
normal body

::: info first
first container body
:::

rest element
```

output2 (wrong):

```
<p>normal body</p>
```

Thanks.

**Comment by koka831** [comment]
The behavior is as expected.
The reason we require a blank line is because the unist-util-visit used in this library interprets directives without blank lines in a certain way, which made parsing very difficult. Therefore, it is necessary to revise the parsing process so that it can handle cases without needing a blank line.

---

```markdown
::: info
first child
<div>second child</div>
:::
```

will be converted in (with some simplify):

    { type: "paragraph", value: "::: info" }
    { type: "paragraph", value: "first child" }
    { type: "html": value: "<div>second child</div>:::" } ← this makes hard to parse

https://github.com/koka831/remark-custom-container/issues/120#issuecomment-1324659675
